stl/inc/iterator

```diff
@@ -324,21 +340,21 @@ public:
         return *this;
     }
 
-    _NODISCARD ostream_iterator& operator*() { // pretend to return designated value
+    _NODISCARD ostream_iterator& operator*() noexcept /* strengthened */ {
         return *this;
     }
 
-    ostream_iterator& operator++() { // pretend to preincrement
+    ostream_iterator& operator++() noexcept /* strengthened */ {
         return *this;
     }
 
-    ostream_iterator& operator++(int) { // pretend to postincrement
+    ostream_iterator& operator++(int) noexcept /* strengthened */ {
         return *this;
     }
 
-protected:
-    const _Elem* _Mydelim; // pointer to delimiter string (NB: not freed)
-    ostream_type* _Myostr; // pointer to output stream
+private:
+    const _Elem* _Mydelim = nullptr; // pointer to delimiter string (NB: not freed)
+    ostream_type* _Myostr = nullptr; // pointer to output stream
 };
 
 // CLASS TEMPLATE istreambuf_iterator

```

Comment:
Nitpick: It seems inconsistent that the `// wrap stream buffer as input iterator` comment is being retained for `istreambuf_iterator` (and similarly for `ostreambuf_iterator`, yet the comments are being removed for `istream_iterator` and `ostream_iterator`). It seems like they should be consistently removed.

Reply:
The `[io]stream_iterator` comments were ever-so-slightly more offensive due to lack of punctuation making them hard to parse; I'll remove the comments from the `buf` flavors as well.